sdk/core/azure-core/src/datetime.cpp

```diff
@@ -2,25 +2,30 @@
 // SPDX-License-Identifier: MIT
 
 #include "azure/core/datetime.hpp"
+#include <azure/core/platform.hpp>
 
 #include <cstdio>
 #include <cstring>
 #include <limits>
 #include <stdexcept>
 
-#ifdef _WIN32
+#ifdef AZ_PLATFORM_WINDOWS
+#ifndef WIN32_LEAN_AND_MEAN
 #define WIN32_LEAN_AND_MEAN
+#endif
+#ifndef NOMINMAX
 #define NOMINMAX
+#endif
 #include <windows.h>
-#else
+#elif defined(AZ_PLATFORM_POSIX)
 #include <sys/time.h>
 #endif
 
 using namespace Azure::Core;
 
 DateTime DateTime::Now()
 {
-#ifdef _WIN32
+#ifdef AZ_PLATFORM_WINDOWS
```

Comment:
In this case, we should add an `#else #error` because otherwise there will be a compiler error that there is no return statement, on other platforms. Or use the alternative approach where this function is only defined for windows/posix, based on our offline discussion. Either is fine.